src/p_mobj.cpp

```diff
@@ -3163,6 +3167,53 @@ void AActor::SetRoll(angle_t r, bool interpolate)
 	}
 }
 
+void AActor::SetThrustFactor(FName dmgtype, fixed_t amount)
+{
+	if (!dmgtype || dmgtype == NAME_None || !stricmp("none", dmgtype) || !stricmp("normal", dmgtype))

```

Comment:
You should be thinking twice when you stricmp an FName.  FNames are case insensitive so comparing to NAME_None is sufficient for the first one and much faster.  There's also NAME_Normal.